corehq/apps/linked_domain/ucr.py

```diff
@@ -63,7 +63,7 @@ def create_linked_ucr(domain_link, report_config_id):
             raise DataSourceConfigurationNotFoundError(_(
                 'The data source referenced by this report could not be found.'
             ))
-    if is_data_registry_report(report_config):
+    if not domain_link.is_remote and is_data_registry_report(report_config):
```

Comment:
What will happen if someone _does_ attempt to remote link a data registry report?  With this change, it'll attempt to proceed, whereas I think it'd be more correct to raise the `RegistryNotAccessible` error that you see a few lines below (maybe with a different message, about it being remote).

As I understand the problem, it's that the `is_data_registry_report` check fails because it includes the line `report_config.config.meta.build.registry_slug`, and `report_config.config` attempts to fetch the datasource config from the DB.  It of course isn't there, because it's a remote server.  However, we do have that config already, in the local variable `datasource`, so we could rework things to instead check `datasource.meta.build.registry_slug` in this workflow to preserve the check.

Reply:
Yeah this is a great point, and sounds pretty straightforward to fix. I'll do that shortly.

Reply:
60837290003ca199ca15dce8aa13e1ebdebc9d6c

Reply:
Whoops forgot to think about other usages of `is_data_registry_report`: d234d429c1dc4b42351d11581c712e8defa9e8d7